### PR TITLE
site_admin/copy_pages.php: Use modern DPDatabase::query, escape, sprintf

### PR DESCRIPTION
**Pull request by bpfoley**
Interestingly, several cases here were incorrectly escaped: they
were already validated projectIDs, and are used as table names.

### CODE REVIEW

**Comment by bpfoley** [comment]
I've tested copying pages from one project to another, and the sandbox is at https://www.pgdp.org/~bfoley/c.branch/cleanup12

File: tools/site_admin/copy_pages.php [hunk 1]
```diff
@@ -651,13 +658,14 @@ function do_stuff( $projectid_, $from_image_, $page_name_handling,
...
         {
             $query = sprintf("
-                      SELECT username FROM user_project_info
-                      WHERE projectid = '%s' AND
-                            iste_$event = 1",
-                    mysqli_real_escape_string(DPDatabase::get_connection(), $projectid_['from'])
+                SELECT username FROM user_project_info
+                WHERE projectid = '%s' AND
+                    iste_%s = 1",
+                DPDatabase::escape($projectid_['from']),
```
Comment (chrismiceli) [inline tools/site_admin/copy_pages.php:664]: didn't you say above this was validated already?

Comment (bpfoley) [reply tools/site_admin/copy_pages.php:664]: Good catch. Yes I did!

File: tools/site_admin/copy_pages.php [hunk 2]
```diff
@@ -282,12 +282,13 @@ function do_stuff( $projectid_, $from_image_, $page_name_handling,
 
         // ----------------------
 
-        $res = mysqli_query(DPDatabase::get_connection(), sprintf("
+        $sql = sprintf("
             SELECT image, fileid
             FROM %s
```
Comment (cpeel) [inline tools/site_admin/copy_pages.php:287]: I would embed the `$projectid` directly in the string and drop the `sprintf()`. That way the sprintf is reserved for escaping values and not the table or column names. https://www.pgdp.net/w/index.php?title=DP_Code_Best_Practices#Column_names_as_variables

Comment (cpeel) [reply tools/site_admin/copy_pages.php:287]: Ditto other places where we build SQL in this file.

Comment (bpfoley) [reply tools/site_admin/copy_pages.php]: Done.

File: tools/site_admin/copy_pages.php [hunk 3]
```diff
@@ -596,25 +593,31 @@ function do_stuff( $projectid_, $from_image_, $page_name_handling,
...
+                'image',
+                'fileid'
+            ],
+            [
+                sprintf("'%s'", DPDatabase::escape($c_dst_image)),
+                sprintf("'%s'", DPDatabase::escape($c_dst_fileid))
+            ],
+            $items_list_template);
+
+        $insert = "INSERT INTO {$projectid_['to']}
```
Comment (cpeel) [inline tools/site_admin/copy_pages.php:606]: Move the `INSERT INTO` onto the next line.

File: tools/site_admin/copy_pages.php [hunk 4]
```diff
@@ -651,13 +654,14 @@ function do_stuff( $projectid_, $from_image_, $page_name_handling,
         foreach ( $subscribable_project_events as $event => $label )
         {
             $query = sprintf("
-                      SELECT username FROM user_project_info
-                      WHERE projectid = '%s' AND
-                            iste_$event = 1",
-                    mysqli_real_escape_string(DPDatabase::get_connection(), $projectid_['from'])
+                SELECT username FROM user_project_info
+                WHERE projectid = '%s' AND
+                    iste_%s = 1",
```
Comment (cpeel) [inline tools/site_admin/copy_pages.php:659]: Keep `$event` in the string since it is a column name. We aren't escaping it or anything anyway.

**Comment by srjfoo** [comment]
I've also tested and didn't see any problems.

**Pull request merged**